docgenerator/src/main/scala/codacy/codenarc/docs/DocGenerator.scala

```diff
@@ -0,0 +1,311 @@
+      paramList => paramList.map(param => Parameter.Specification(Parameter.Name(param), Parameter.Value("")))
+    )
+
+  def parameterDescriptionFromParametersStringList(
+      parameters: Option[Set[String]]
+  ): Option[Set[Parameter.Description]] =
+    parameters.map(
+      paramList => paramList.map(param => Parameter.Description(Parameter.Name(param), Parameter.DescriptionText("")))
+    )
+
+  /**
+    * returns the regex to get information from markdown file for a rule
+    *
+    * @param ruleName Name of the rule to get information
+    * @return
+    */
+  def ruleInfoRegex(ruleName: String) = new Regex("## " + ruleName + "([\\n\\S\\s\\w\\d][^##])*")
+
+  /**
+    * Gets the tool version from the pattern.json file
+    *
+    * @param args
+    * @return
+    */
+  def versionFromArgs(args: Array[String]): String =
+    args.headOption
+      .orElse {
+        val sourceFile = io.Source.fromFile(s"$codeNarcVersionFile")
+        val resourceContent = sourceFile.getLines.mkString("")
+        sourceFile.close()
+        Some(resourceContent)
+      }
+      .getOrElse {
+        throw new Exception("No version provided")
+      }
+
+  /**
+    * Gets the documentation markdown name from the href element value
+    *
+    * @param href
+    * @return
+    */
+  def extractMdFilenameFromHref(href: String): String =
+    href.replaceAllLiterally("./", "").replaceAllLiterally(".html", "").split("#")(0) + ".md"
+
+  /**
+    * Get information about a rule from markdown file
+    *
+    * @param docsFolder   Documentation folder
+    * @param ruleName     Name of the rule to get information about
+    * @param ruleFilename The name of the markdown file where rule information is stored at.
+    * @return
+    */
+  def getRuleExtendedInfoFromMarkdown(docsFolder: String, ruleName: String, ruleFilename: String): String = {
+    val ruleInfoMarkdown = File(s"$docsFolder$ruleFilename").contentAsString
+
+    val regexPattern = ruleInfoRegex(ruleName)
+    val regexMatch = regexPattern.findFirstIn(ruleInfoMarkdown).getOrElse("")
+    regexMatch.substring(regexMatch.indexOf('\n') + 1) // remove the title
+  }
+
+  /**
+    * Get rule simplified description from codenarc-base-messages.properties file
+    *
+    * @param directory CodeNarc base directory
+    * @param patternId rule/pattern name
+    * @return Simplified Description
+    */
+  def getRuleBasicDescription(directory: String, patternId: String): String = {
+    // This method trims the description to a max length of 500.
+    // The basic description has a limit of 500 chars, so it is required to trim if it is longer than that
+    // The trim is being done by removing a sentence from the end each time until it has less than 500 length.
+    @tailrec
+    def descriptionTrimAux(description: String): String = {
+      if (description.length < 500) {
+        description
+      } else {
+        descriptionTrimAux(description.substring(0, description.lastIndexOf(".")))
+      }
```

Comment:
Avoid braces in if else: https://docs.scala-lang.org/style/control-structures.html
```scala
def descriptionTrimAux(description: String): String =
  if (description.length >= 500) descriptionTrimAux(description.substring(0, description.lastIndexOf(".")))
  else description
```